### PR TITLE
wix-ui-core: Allow setting toggle icon through prop 

### PR DESCRIPTION
**Pull request by ronenst**

### CODE REVIEW

File: packages/wix-ui-core/src/components/ToggleSwitch/ToggleSwitch.tsx [hunk 1]
```diff
@@ -18,13 +17,20 @@ export type ToggleSwitchStyles = {
...
+};
+
 export interface ToggleSwitchProps {
   checked?: boolean;
   disabled?: boolean;
   onChange?: React.EventHandler<React.ChangeEvent<HTMLInputElement>>;
   classes?: ToggleSwitchClasses;
   styles?: ToggleSwitchStyles;
   id?: string;
+  IconComponent?: React.ComponentType<ToggleSwitchIconProps>;
```
Comment (argshook) [inline packages/wix-ui-core/src/components/ToggleSwitch/ToggleSwitch.tsx]: no need for this, `iconOn` `iconOff` should be enough

File: packages/wix-ui-core/src/components/ToggleSwitch/ToggleSwitch.tsx [hunk 2]
```diff
@@ -52,7 +58,12 @@ class ToggleSwitch extends React.PureComponent<ToggleSwitchProps> {
     id: string,
   };
 
-  static defaultProps = {checked: false, styles: {}};
+  static defaultProps = {
+    checked: false,
+    styles: {},
+    iconOn: <div>{'✓'}</div>,
```
Comment (argshook) [inline packages/wix-ui-core/src/components/ToggleSwitch/ToggleSwitch.tsx]: should be using something from `wix-ui-icons-common` (which conveniently doesn't yet exist, but will)

File: packages/wix-ui-core/src/components/ToggleSwitch/ToggleSwitch.tsx [hunk 3]
```diff
@@ -76,9 +87,15 @@ class ToggleSwitch extends React.PureComponent<ToggleSwitchProps> {
...
 
+  defaultIconComponent = ({checked}) => {
+    const {iconOn, iconOff} = this.props;
+    return checked ? iconOn : iconOff;
+  }
+
   render() {
     const {checked, disabled, classes, styles} = this.props;
     const {id} = this;
+    const IconComponent = this.props.IconComponent || this.defaultIconComponent;
```
Comment (argshook) [inline packages/wix-ui-core/src/components/ToggleSwitch/ToggleSwitch.tsx]: `IconComponent` is not needed if `iconOn` & `iconOff` is used directly

File: packages/wix-ui-core/src/components/ToggleSwitch/ToggleSwitch.tsx [hunk 4]
```diff
@@ -92,9 +109,9 @@ class ToggleSwitch extends React.PureComponent<ToggleSwitchProps> {
 
         <label htmlFor={id} className={classes.outerLabel} style={styles.outerLabel}/>
         <label htmlFor={id} className={classes.innerLabel} style={styles.innerLabel}>
-          <svg className={classes.toggleIcon} style={styles.toggleIcon} viewBox={getViewBox(checked)}>
-            <path d={getPathDescription(checked)}/>
-          </svg>
+          <div className={classes.toggleIcon} style={styles.toggleIcon}>
+            <IconComponent checked={checked}/>
```
Comment (argshook) [inline packages/wix-ui-core/src/components/ToggleSwitch/ToggleSwitch.tsx]: `checked ? iconOn : iconOff`

**Comment by ronenst** [comment]
Build failing. Requires https://github.com/wix/wix-ui/pull/159

**Comment by shlomitc** [comment]
Icons don't appear in storybook on the ToggleSwitch

**Comment by ronenst** [comment]
This is consistent with the current (master) behavior and is due to the default theme which has it hidden, what theme should be used now for it?

**Comment by ronenst** [comment]
I've added something simple, PTAL

**Comment by shlomitc** [comment]
Closing this PR since stylable version of this component should be the only relevant one (and icons will be solved there)

**Pull request closed without merging**